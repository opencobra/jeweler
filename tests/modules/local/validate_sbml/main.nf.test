nextflow_process {

    name 'Test Process VALIDATE_SBML'
    script 'modules/local/validate_sbml/main.nf'
    process 'VALIDATE_SBML'
    tag 'modules'
    tag 'modules_local'
    tag 'validate_sbml'

    test('Should positively validate a good SBML document') {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.success

            with(process.out.report) {
                assert size() == 1

                with(get(0)) {
                    assert get(1).strip() == 'valid'
                    // FIXME: Accessing compressed JSON content is currently not supported,
                    //  see https://github.com/askimed/nf-test/issues/156
                    // def expectedKeys = [
                        // 'SBML_FATAL',
                        // 'SBML_ERROR',
                        // 'SBML_SCHEMA_ERROR',
                        // 'SBML_ERROR',
                        // 'SBML_WARNING',
                        // 'COBRA_FATAL',
                        // 'COBRA_ERROR',
                        // 'COBRA_WARNING',
                        // 'COBRA_CHECK'
                    // ] as Set
                    // assert path(get(2)).json.keys.intersect(expectedKeys) == expectedKeys
                    assert snapshot(path(get(2))).match()
                }
            }

            // Reported versions should match snapshots.
            assert snapshot(process.out.versions.first()).match('versions')
        }

    }

    test('Should negatively validate a bad SBML document') {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_invalid'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_invalid.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.success

            with(process.out.report) {
                assert size() == 1

                with(get(0)) {
                    assert get(1).strip() == 'invalid'
                    // FIXME: Accessing compressed JSON content is currently not supported,
                    //  see https://github.com/askimed/nf-test/issues/156
                    // def expectedKeys = [
                        // 'SBML_FATAL',
                        // 'SBML_ERROR',
                        // 'SBML_SCHEMA_ERROR',
                        // 'SBML_ERROR',
                        // 'SBML_WARNING',
                        // 'COBRA_FATAL',
                        // 'COBRA_ERROR',
                        // 'COBRA_WARNING',
                        // 'COBRA_CHECK'
                    // ] as Set
                    // assert path(get(2)).json.keys.intersect(expectedKeys) == expectedKeys
                    assert snapshot(path(get(2))).match()
                }
            }

            // Reported versions should match snapshots.
            assert snapshot(process.out.versions.first()).match('versions')
        }

    }
}
