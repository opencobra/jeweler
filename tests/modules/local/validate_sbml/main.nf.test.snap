{
    "versions": {
        "content": [
            "versions.yml:md5,aee549eaafd4530ffdde074a8ac002ee"
        ],
        "timestamp": "2023-11-28T14:51:33.170129371"
    },
    "Should positively validate a good SBML document": {
        "content": [
            "e_coli_core.json.gz:md5,0e7d8540d9ab428f56193ef7336b3fe6"
        ],
        "timestamp": "2023-11-28T15:48:34.956083266"
    },
    "Should negatively validate a bad SBML document": {
        "content": [
            "e_coli_invalid.json.gz:md5,050e3669abbec8adba4128aff70156cf"
        ],
        "timestamp": "2023-11-28T15:50:43.497738208"
    }
}